### PR TITLE
initial import of zookeeper client

### PR DESCRIPTION
**Pull request by eishay**
from https://github.com/twitter/scala-zookeeper-client/blob/master/src/main/scala/com/twitter/zookeeper/ZooKeeperClient.scala

### CODE REVIEW

**Comment by andrewconner** [comment]
Ooo. I'm _really_ excited about this.

File: shoebox/app/com/keepit/common/zookeeper/ZooKeeperClient.scala [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+import org.apache.zookeeper.data.{ACL, Stat, Id}
+import org.apache.zookeeper.ZooDefs.Ids
+import org.apache.zookeeper.Watcher.Event.EventType
+import org.apache.zookeeper.Watcher.Event.KeeperState
+import java.util.concurrent.{CountDownLatch, TimeUnit}
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * The code was originally taken from https://github.com/twitter/scala-zookeeper-client/blob/master/src/main/scala/com/twitter/zookeeper/ZooKeeperClient.scala
+ * It was banded by twitter in favor of https://github.com/twitter/util/tree/master/util-zk
```
Comment (gmethvin) [inline shoebox/app/com/keepit/common/zookeeper/ZooKeeperClient.scala]: banded?

Comment (eishay) [inline shoebox/app/com/keepit/common/zookeeper/ZooKeeperClient.scala]: short for abandoned ;-)

File: shoebox/app/com/keepit/common/zookeeper/ZooKeeperClient.scala [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+/**
+ * The code was originally taken from https://github.com/twitter/scala-zookeeper-client/blob/master/src/main/scala/com/twitter/zookeeper/ZooKeeperClient.scala
+ * It was abandoned by twitter in favor of https://github.com/twitter/util/tree/master/util-zk
+ */
+class ZooKeeperClient(servers: String, sessionTimeout: Int, basePath : String,
+                      watcher: Option[ZooKeeperClient => Unit]) extends Logging {
+  @volatile private var zk : ZooKeeper = null
+  connect()
+
+  def this(servers: String, sessionTimeout: Int, basePath : String) =
```
Comment (gmethvin) [inline shoebox/app/com/keepit/common/zookeeper/ZooKeeperClient.scala:24]: Alternate constructors really aren't Scala style. Maybe it was designed this way to work better alongside Java code?

**Comment by gmethvin** [comment]
Awesome.

**Comment by 2is10** [comment]
Bring on the lions, tigers, and bears!

File: shoebox/app/com/keepit/common/zookeeper/ZooKeeperClient.scala (hunk 2 above)
Comment (gmethvin) [inline shoebox/app/com/keepit/common/zookeeper/ZooKeeperClient.scala:24]: feel free not to change this though

Comment (eishay) [inline shoebox/app/com/keepit/common/zookeeper/ZooKeeperClient.scala:24]: I'm going to do many changes to this code to make it fit our needs. These constructors will probably be kicked out, its an initial import

**Pull request merged**